src/clj/nnts2/server.clj

```diff
@@ -10,36 +10,43 @@
             [nnts2.middleware :refer [wrap-kebab-case not-found wrap-exception-handling wrap-log-request-response wrap-validate-access-token]]
             [ring.middleware.params :refer [wrap-params]]
             [ring.middleware.session :refer [wrap-session]]
-            [ring.middleware.json :refer [wrap-json-response]]
+            [ring.middleware.json :refer [wrap-json-response  wrap-json-body]]
             [nnts2.config :refer [oauth2-spec]]
             [ring.middleware.defaults :refer [site-defaults wrap-defaults]]
             [ring.middleware.resource :as resource]
             [ring.middleware.session.memory :as mem]
+            [nnts2.user.middleware :refer [wrap-nnts-user-id]]
             [nnts2.user.routes :as user]
+            [nnts2.note.routes :as note]
             [ring.middleware.cookies :as cookies]
             [compojure.response :refer [render]]
             [clojure.java.io :as io]
             [ring.util.response :refer [file-response resource-response
                                         status content-type]]))
 
+
 (defonce ^:private all-sessions (mem/memory-store))
 
+(def auth-routes (compojure.core/routes
+                   user/routes
+                   (context "/note" [] note/routes)))
+
 (defroutes app-routes
```

Comment:
The convention is to move all routes to `routes.clj` and to simply define the middlewares and the server lifecycle methods here.

Reply:
ok, we were thinking of separating routes requiring authentication from the ones  which wont.
And a few middleware are only used on the auth-routes. What will be your suggestion for that?

Reply:
You could still keep them in `routes.clj`  as auth and non-auth routes and then combine them to form the final `routes`.